shell/components/formatter/LinkName.vue

```diff
@@ -41,15 +42,29 @@ export default {
       };
 
       return { name, params };
+    },
+
+    hasPermission() {
+      if (this.type === NODE_TYPE) {
```

Comment:
Think we can make this generic. In `shell/middleware/authenticated` there's an `invalidResource` function which is used to determine if a schema for a type exists

```
  // Note - don't use the current products store... because products can override stores for resources with `typeStoreMap`
  const inStore = store.getters['currentStore'](resource);
  // There's a chance we're in an extension's product who's store could be anything, so confirm schemaFor exists
  const schemaFor = store.getters[`${ inStore }/schemaFor`];

  // In order to check a resource is valid we need these
  if (!inStore || !schemaFor) {
    return false;
  }

  // Resource is valid if a schema exists for it (standard resource, spoofed resource) or it's a virtual resource
  const validResource = schemaFor(resource) || store.getters['type-map/isVirtual'](resource);

  if (validResource) {
    return false;
  }
```

That part can be made into a function, lets go with canViewResource in `shell/utils/auth.js`, and used there and in this file.

There are other checks we can do to determine if the resource is available (like if the schema collection and resource links provide a way for the user to fetch it) but just checking the schema is there is ok for now

Reply:
Thanks for the review, I've changed the code